rest_framework/serializers.py

```diff
@@ -947,6 +947,11 @@ def get_fields(self):
         # Determine any extra field arguments and hidden fields that
         # should be included
         extra_kwargs = self.get_extra_kwargs()
+        for field in extra_kwargs:
+            assert field not in declared_fields, (
+                "Field {} is declared on the class and also present in "
+                "extra_kwargs or read_only_fields".format(field)
+            )
```

Comment:
I use this functionality all the time, to override field attributes on inherited serializers.

I would even argue to leave the current behavior to allow providing extra kwargs, that is handy for providing common kwargs to multiple fields by a simple loop, (though, it can be done by overriding __init__, I had preferred this one).